src/mixer/basetrackplayer.cpp

```diff
@@ -105,6 +105,36 @@ BaseTrackPlayerImpl::BaseTrackPlayerImpl(
     m_pTrackColor->connectValueChangeRequest(
             this, &BaseTrackPlayerImpl::slotTrackColorChangeRequest);
 
+    m_pTrackColorPrev = std::make_unique<ControlPushButton>(
+            ConfigKey(getGroup(), "track_color_prev"));
+    connect(m_pTrackColorPrev.get(),
+            &ControlPushButton::valueChanged,
+            this,
+            [this](double value) {
+                if (value > 0)
+                    BaseTrackPlayerImpl::slotTrackMoveColor(-1);
+            });
+
+    m_pTrackColorNext = std::make_unique<ControlPushButton>(
+            ConfigKey(getGroup(), "track_color_next"));
+    connect(m_pTrackColorNext.get(),
+            &ControlPushButton::valueChanged,
+            this,
+            [this](double value) {
+                if (value > 0)
+                    BaseTrackPlayerImpl::slotTrackMoveColor(1);
+            });
+
+    m_pTrackColorSelect = std::make_unique<ControlEncoder>(
+            ConfigKey(getGroup(), "track_color_selector"), false);
+    connect(m_pTrackColorSelect.get(),
+            &ControlEncoder::valueChanged,
```

Comment:
The ControlEcoder header needs to be included, the forward-declartion of the class in the header is not sufficient.
#include "control/controlencoder.h"

Reply:
This is weird as it compiles well on my compoteur (debian bookworm) with same kit and target as the build server
That might be a configuration issue on my side, do you have any guideline on this ?

Reply:
And controlPushButton neither have forward-declaration or header included, i'm kind of surprised

Reply:
I think ControlPushButton is included indirectly via `engine/enginemixer.h` or another include.

I remeber a similiar issue (the other way around) where it wouldn't build locally but CI had pre-compiled headers enabled and could build.

Just include controlencoder, problem solved.